plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/parameterInfo/KotlinHighLevelFunctionParameterInfoHandler.kt

```diff
@@ -191,6 +191,20 @@ abstract class KotlinHighLevelParameterInfoWithCallHandlerBase<TArgumentList : K
                     }
                 }
 
+                var hasTypeMismatchBeforeCurrent = false
+                for ((index, argument) in arguments.withIndex()) {
+                    if (index >= currentArgumentIndex) break
+                    val parameterForArgument = argumentMapping[argument] ?: continue
+                    if (parameterForArgument == setValueParameter) continue
+
+                    val argumentType = argument.getKtType() ?: error("Argument should have a KtType")
+                    val parameterType = parameterForArgument.annotatedType.type
+                    if (argumentType.isNotSubTypeOf(parameterType)) {
```

Comment:
Does this work for the following code? (I am having some build issue locally and so far have not been able to test this 😢)
```
fun <T> takeT(t: MutableList<T>, i: Int) {
}

fun get(): MutableList<out String> {
    return mutableListOf()
}

fun test() {
    takeT(get(), <caret>)
}
```

Reply:
No it does not work correctly. I guess we need to substitute the type parameters then. :slightly_smiling_face: